File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/process/prinfo/vb/PrInfo.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Diagnostics
Imports System.Threading

Namespace PrInfo

    Module PrInfo

        Public Sub Main()

            Dim args As String()
            Dim appName As String
            args = Environment.GetCommandLineArgs()
            appName = args(0)

            If (args.Length <> 3 And (args.Length <> 2 OrElse args(1) <> "e")) Then
                Console.WriteLine("Usage: ")
                Console.WriteLine("    " + appName + " e: Enumerate processes")
                Console.WriteLine("    " + appName + " <command> <process_id>")
                Console.WriteLine("")
                Console.WriteLine("         <command> =")
                Console.WriteLine("            i: Show process info")
                Console.WriteLine("            c: Close")
                Console.WriteLine("            k: Kill")
                Console.WriteLine("            p: show curent priority")
                Exit Sub
            End If

            Dim command As String
            command = args(1)

            If (command = "e") Then
                Dim processes() As Process
                processes = Process.GetProcesses()

                Dim myprocess As Process
                Dim i As Integer
                For i = 0 To processes.length - 1
                    myprocess = processes(i)
                    Console.WriteLine(CStr(myprocess.Id) + "    : " + myprocess.ProcessName)
                Next
            Else
                Dim processid As int32
                processid = Int32.Parse(args(2))
                Dim myprocess As Process
                myprocess = Process.GetProcessById(processid)

                Select Case (command)
                    Case "c"
                        myprocess.CloseMainWindow()
                    case "k":
                        myprocess.Kill()
                    case "p":
                        Console.WriteLine("Priority: " + CStr(myprocess.PriorityClass))
                    case "i":
                        Console.WriteLine("Priority Class         :{0}", CStr(myprocess.PriorityClass))
                        Console.WriteLine("Handle Count           :{0}", myprocess.HandleCount)
                        Console.WriteLine("Main Window Title      :{0}", myprocess.MainWindowTitle)
                        Console.WriteLine("Min Working Set        :{0}", myprocess.MinWorkingSet)
                        Console.WriteLine("Max Working Set        :{0}", myprocess.MaxWorkingSet)
                        Console.WriteLine("Paged Memory Size      :{0}", myprocess.PagedMemorySize)
                        Console.WriteLine("Peak Paged Memory Size :{0}", myprocess.PeakPagedMemorySize)
                End Select
            End If
        End Sub
    End Module
End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/process/prinfo/cs/prinfo.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Diagnostics;
using System.Threading;

public class ProcessInfo
{
    public static void Main(String[] args)
    {
        string appName = Environment.GetCommandLineArgs()[0];

        if(args.Length != 2 && (args.Length != 1 || args[0]!="e"))
        {
            Console.WriteLine("Usage: ");
            Console.WriteLine("\t" + appName +" e: Enumerate processes");
            Console.WriteLine("\t" + appName +" <command> <process_id>");
            Console.WriteLine("");
            Console.WriteLine("\t\t<command> =");
            Console.WriteLine("\t\t\ti: Show process info");
            Console.WriteLine("\t\t\tc: Close");
            Console.WriteLine("\t\t\tk: Kill");
            Console.WriteLine("\t\t\tp: show curent priority");
            return;
        }

        string command = args[0];

        if(command =="e")
        {
            Process[] processes = Process.GetProcesses();

            foreach(Process process in processes)
            {
                Console.WriteLine(process.Id + "\t: " + process.ProcessName);
            }
        }
        else
        {

            Int32 processid = Int32.Parse(args[1]);
            Process process = Process.GetProcessById(processid);

            switch (command)
            {
            case "c":
                process.CloseMainWindow();
                break;
            case "k":
                process.Kill();
                break;
            case "p":
                Console.WriteLine("Priority: "+ process.PriorityClass.ToString("G"));
                break;
            case "i":
                Console.WriteLine("Priority Class         :{0}", process.PriorityClass.ToString("G"));
                Console.WriteLine("Handle Count           :{0}", process.HandleCount);
                Console.WriteLine("Main Window Title      :{0}", process.MainWindowTitle);
                Console.WriteLine("Min Working Set        :{0}", process.MinWorkingSet);
                Console.WriteLine("Max Working Set        :{0}", process.MaxWorkingSet);
                Console.WriteLine("Paged Memory Size      :{0}", process.PagedMemorySize);
                Console.WriteLine("Peak Paged Memory Size :{0}", process.PeakPagedMemorySize);
                break;
            }


        }
    }
}

